plugins/redis/redis.cpp

```diff
@@ -180,26 +202,35 @@ static void releaseContext()
     }
     threadHooked = false;
 }
-Connection::Connection(ICodeContext * ctx, const char * _options, unsigned __int64 _database, const char * password, unsigned __int64 _timeout)
+Connection::Connection(ICodeContext * ctx, const char * _options, int _database, const char * password, unsigned _timeout)
   : database(0), timeout(_timeout), port(0), serverIpPortPasswordHash(hashServerIpPortPassword(ctx, _options, password))
 {
     options.set(_options, strlen(_options));
     parseOptions(ctx, _options);
     connect(ctx, _database, password);
 }
-Connection::Connection(ICodeContext * ctx, const char * _options, const char * _ip, int _port, unsigned _serverIpPortPasswordHash, unsigned __int64 _database, const char * password, unsigned __int64 _timeout)
+Connection::Connection(ICodeContext * ctx, const char * _options, const char * _ip, int _port, unsigned _serverIpPortPasswordHash, int _database, const char * password, unsigned _timeout)
   : database(0), timeout(_timeout), serverIpPortPasswordHash(_serverIpPortPasswordHash), port(_port)
 {
     options.set(_options, strlen(_options));
     ip.set(_ip, strlen(_ip));
     connect(ctx, _database, password);
 }
-void Connection::connect(ICodeContext * ctx, unsigned __int64 _database, const char * password)
+void Connection::redisConnect()
 {
-    struct timeval to = { timeout/1000, (timeout%1000)*1000 };
-    context = redisConnectWithTimeout(ip.str(), port, to);
-    assertConnection();
-    redisSetTimeout(context, to);
+    unsigned _timeLeft = timeLeft();
+    if (_timeLeft == 0)
```

Comment:
@ghalliday  It's the same thing, only without the extra TimeoutHandler::getTimeout() in the nominal case where timeouts are used. 
Connection::timeLeft() now only returns 0 if timeout = 0, otherwise it either fails ( timeout > 0 && TImeoutHandler::timeLeft() > timeout) or returns the time left on the counter ( timeout > 0 && TImeoutHandler::timeLeft() < timeout). Please check that this is true and looks ok.